### DISCUSSION TITLE
How do I get VS Code Typescript Intellisense to recognize an alias that is resolved after the project is compiled?

### DISCUSSION THREAD

**Discussion by thoughtassassin** [parcel-bundler/parcel#7807] [Q&A] [answered]
I have a directory called wa-shared in a monorepo. The files are copied from that directory into any one of the apps at the time that app is compiled. The folder structure looks sort of like:

Project Directory
  wa-shared
  app1
  app2
  app3

So if I run app1, the contents of wa-shared are copied into app1 and the project is compiled. I have an alias in the package.json file:

"allas": {
  ...
 "wa-shared": "./src/path/to/shared"
 ...
}
 
If I run the project TypeScript and Parcel recognizes and compiles the file just fine. I think the problem is that before it is compiled VS Code Intellisense has no way of knowing that file will be copied into the local directory. Therefore, the import in VS Code shows a resolution error:

_Cannot find module 'wa-shared/src/assets/img/CrossX' or its corresponding type declarations.ts(2307)_

Parcel recognizes this just fine and compiles the code. Is there any way to make VS Code recognize that and not flag that line as an error?

**Comment by thoughtassassin** [comment]
It does look like a VS Code issue. I was able to get VS Code to recognize the path with this .tsconfig:

```{
  "compilerOptions": {
    "sourceMap": true,
    "allowJs": true,
    "esModuleInterop": true,
    "jsx": "react",
    "baseUrl": ".",
    "paths": {
      "wa-shared/*": [
        "./shared/*"
      ]
    }
  },
  "exclude": [
    "node_modules",
    "cypress",
  ]
}
```
However, now Parcel is throwing this error when it tries to compile:

`@parcel/transformer-js: Unexpected token `[`. Expected jsx identifier`

```
    10 |         <reactstrap_1.Button aria-labelledby={"remove".concat(domain)} className="remove" onClick={f
  > 11 |             <CrossX_1["default"] fill="#A22121"/>
  >    |                      ^
    12 |             <label id={"remove".concat(domain)} className="sr-only">
    13 |                 Remove
```

BTW this is NOT an answer. I wanted to add an update to the quesiton.

**Comment by thoughtassassin** [comment, marked as answer]
I was working on this all day and I discovered that I had a .js file in addition .tsx file. That is what was causing the new error. It turns out that all I need was the glob pattern in my paths in the tsconfig file:

```
...
"baseUrl": ".",
"paths": {
      "wa-shared/*": [
        "./shared/*"
      ]
    }
...
```

**Comment by lunarnet76** [comment]
it did not worked for me however specifying the type explicitely did work

"@/*": ["./src/*", "./src/*.tsx", "./src/*.ts"],
            "@hooks/*": ["./src/hooks/*", "./src/hooks/*.tsx", "./src/hooks/*.ts"],
            "@components/*": ["./src/components/*", "./src/components/*.tsx", "./src/components/*.ts"],
